examples/system/cpp_pthread/main/cpp_pthread.cpp

```diff
@@ -0,0 +1,109 @@
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include <memory>
+#include <string>
+#include <sstream>
+#include <esp_pthread.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+#include <esp_log.h>
+
+using namespace std::chrono;
+
+const auto sleep_time = seconds{5};
+
+void print_thread_info(const char* extra = nullptr)
+{
+        std::stringstream ss;
+        if(extra)
+        {
+            ss << extra;
+        }
+        ss << "Core id: " << xPortGetCoreID()
+            << ", prio: " << uxTaskPriorityGet(nullptr) 
+            << ", minumim free stack: " << uxTaskGetStackHighWaterMark(nullptr) << " bytes.";
+        ESP_LOGI(pcTaskGetTaskName(nullptr), "%s", ss.str().c_str());        
+}
+
+void thread_func_inherited()
+{
+    while(true)
+    {
+        print_thread_info("This is the INHERITING thread with the same parameters as our parent, including name. ");
+        std::this_thread::sleep_for(sleep_time);
+    }
+}
+
+void spwan_another_thread()
+{
+    // Create a new thread, it will inherit our configuration
+    std::thread inherits(thread_func_inherited);
+
+    while(true)
+    {
+        print_thread_info();
+        std::this_thread::sleep_for(sleep_time);
+    }
+}
+
+void thread_func_any_core()
+{
+    while(true)
+    {
+        print_thread_info("This thread may run on any core. ");
+        std::this_thread::sleep_for(sleep_time);
+    }
+}
+
+void thread_func()
+{
+    while(true)
+    {
+        print_thread_info();
+        std::this_thread::sleep_for(sleep_time);
+    }
+}
+
+auto create_config(const char* name, int core_id, int stack, int prio)
+{
+    auto cfg = esp_create_default_pthread_config();
+    esp_set_pthread_name(&cfg, name);
+    cfg.pin_to_core = core_id;
+    cfg.stack_size = stack;
+    cfg.prio = prio;
+    return cfg;
+}
+
+extern "C"
+{
+    void app_main()
```

Comment:
Can you please make it `extern "C" void app_main()` to save one indentation level?

Reply:
Yes, silly me.